pkg/splunk/client/azureblobclient.go

```diff
@@ -108,126 +203,242 @@ func NewAzureBlobClient(ctx context.Context, bucketName string, storageAccountNa
-		downloadRequest.LocalFile, downloadRequest.Etag)
+	scopedLog := reqLogger.WithName("AzureBlob:extractResponse")
+
+	azureAppsRemoteData := RemoteDataListResponse{}
 
-	file, err := os.Create(downloadRequest.LocalFile)
+	// Read response body
+	responseDownloadBody, err := ioutil.ReadAll(httpResponse.Body)
 	if err != nil {
-		scopedLog.Error(err, "Unable to create local file")
-		return false, err
+		scopedLog.Error(err, "Azure blob,Errored when reading resp body for app download")
 	}
-	defer file.Close()
-
-	azureBlobClient := client.Client
-
-	//TODO : revise in next sprint . these are just place holder to more
-	// specific data structure so we can pass download options to rest calls
-	downloadOptions := make(map[string]string)
 
-	downloadOptions["maxChunkSize"] = "5gb"
-	downloadOptions["timeout"] = "5sec"
+	// Variable to hold unmarshaled data
+	data := &EnumerationResults{}
 
-	err = azureBlobClient.DownloadApp(ctx, client.BucketName, downloadRequest.RemoteFile, downloadRequest.LocalFile, downloadOptions)
+	// Unmarshal http response
+	err = xml.Unmarshal([]byte(responseDownloadBody), data)
 	if err != nil {
-		scopedLog.Error(err, "Unable to download remote file")
-		return false, err
+		scopedLog.Error(err, "Errored  unmarshalling app packages list")
+		return azureAppsRemoteData, err
 	}
 
-	scopedLog.Info("File downloaded")
+	// Extract data from all blobs
+	for count := 0; count < len(data.Blobs.Blob); count++ {
+		// Extract blob
+		blob := data.Blobs.Blob[count]
+
+		scopedLog.Info("Listing App package details", "Count:", count, "App package name:", blob.Name, "Etag:", blob.Properties.ETag,
+			"Created on:", blob.Properties.CreationTime, "Modified on:",
+			blob.Properties.LastModified, "Content Size:", blob.Properties)
+
+		// Extract properties
+		newETag := blob.Properties.ETag
+		newKey := blob.Name
+		newLastModified, errTime := time.Parse(http.TimeFormat, blob.Properties.LastModified)
+		if errTime != nil {
+			scopedLog.Error(err, "Unable to get lastModifiedTime, not adding to list", "App Package", newKey, "name", blob.Properties.LastModified)
+			continue
+		}
+		newSize, errInt := strconv.ParseInt(blob.Properties.ContentLength, 10, 64)
+		if errInt != nil {
+			scopedLog.Error(err, "Unable to get newSize, not adding to list", "App package", newKey, "name", blob.Properties.ContentLength)
+			continue
+		}
+		newStorageClass := "standard" //TODO : map to a azure blob field
+
+		// Create new object and append
+		newRemoteObject := RemoteObject{Etag: &newETag, Key: &newKey, LastModified: &newLastModified, Size: &newSize, StorageClass: &newStorageClass}
+		azureAppsRemoteData.Objects = append(azureAppsRemoteData.Objects, &newRemoteObject)
+	}
+
+	return azureAppsRemoteData, nil
+}
 
+// DownloadApp downloads an app package from remote storage
+func (client *AzureBlobClient) DownloadApp(ctx context.Context, downloadRequest RemoteDataDownloadRequest) (bool, error) {
```

Comment:
Future work?

Reply:
Yes, we will be implementing download in the next sprint.